netty/src/main/java/io/grpc/transport/netty/NettyServerHandler.java

```diff
@@ -287,6 +289,12 @@ private void sendResponseHeaders(ChannelHandlerContext ctx, SendResponseHeadersC
     encoder().writeHeaders(ctx, cmd.streamId(), cmd.headers(), 0, cmd.endOfStream(), promise);
   }
 
+  private void cancelStream(ChannelHandlerContext ctx, CancelServerStreamCommand cmd,
+      ChannelPromise promise) {
+    serverStream(requireHttp2Stream(cmd.streamId())).abortStream(cmd.reason(), false);
```

Comment:
@ejona86 It turns out that calling `abortStream(status, true)` will enqueue a `SendResponseHeadersCommand` to be processed by the stream that we're about to make non-writeable, which doesn't work. :) I think the completely correct thing would be to make `abortStream` optionally synchronous by returning a future (or passing a promise). For now though, not notifying the client of the status seems fine since there's a TODO to remove that option altogether.